File: program5.c
#include <stdio.h>
#define SIZE 5

int stack[SIZE], top = -1;

void push(int value) {
    if (top == SIZE - 1)
        printf("Stack Overflow\n");
    else {
        top++;
        stack[top] = value;
        printf("%d pushed to stack\n", value);
    }
}

void pop() {
    if (top == -1)
        printf("Stack Underflow\n");
    else {
        printf("%d popped from stack\n", stack[top]);
        top--;
    }
}

void peek() {
    if (top == -1)
        printf("Stack is Empty\n");
    else
        printf("Top element is %d\n", stack[top]);
}

void display() {
    if (top == -1)
        printf("Stack is Empty\n");
    else {
        printf("Stack elements: ");
        for (int i = top; i >= 0; i--)
            printf("%d ", stack[i]);
        printf("\n");
    }
}

int main() {
    int choice, value;

    while (1) {
        printf("\n1.PUSH\n2.POP\n3.PEEK\n4.DISPLAY\n5.EXIT\nEnter choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter value to push: ");
                scanf("%d", &value);
                push(value);
                break;
            case 2:
                pop();
                break;
            case 3:
                peek();
                break;
            case 4:
                display();
                break;
            case 5:
                return 0;
            default:
                printf("Invalid choice\n");
        }
    }
}

File: program5.py
# program5.py
# Stack operations: PUSH, POP, PEEK

stack = []  # using Python list as stack

def push():
    element = input("Enter element to PUSH: ")
    stack.append(element)
    print(f"{element} pushed to stack")

def pop_element():
    if not stack:
        print("Stack is empty! Cannot POP.")
    else:
        print(f"Element popped: {stack.pop()}")

def peek():
    if not stack:
        print("Stack is empty! No element to PEEK.")
    else:
        print(f"Top element: {stack[-1]}")

def display():
    print("Current Stack:", stack)

while True:
    print("\n--- STACK OPERATIONS ---")
    print("1. PUSH")
    print("2. POP")
    print("3. PEEK")
    print("4. DISPLAY STACK")
    print("5. EXIT")

    choice = input("Enter your choice: ")

    if choice == '1':
        push()
    elif choice == '2':
        pop_element()
    elif choice == '3':
        peek()
    elif choice == '4':
        display()
    elif choice == '5':
        print("Exiting program...")
        break
    else:
        print("Invalid choice! Try again.")
